src/lib/support/PrivateHeap.cpp

```diff
@@ -267,6 +269,42 @@ extern "C" void PrivateHeapFree(void * ptr)
     }
 }
 
+void * PrivateHeapRealloc(void * heap, void * ptr, size_t size)
+{
+    if (ptr == nullptr)
+    {
+        return PrivateHeapAlloc(heap, size);
+    }
+
+    if (size == 0)
+    {
+        PrivateHeapFree(ptr);
+        return nullptr;
+    }
+
+    PrivateHeapBlockHeader * header =
+        reinterpret_cast<PrivateHeapBlockHeader *>(static_cast<char *>(ptr) - sizeof(PrivateHeapBlockHeader));
+
+    ValidateHeader(header);
+
+    if (header->nextBytes >= size)
+    {
+        return ptr; // no reallocation needed
+    }
+
+    void * largerCopy = PrivateHeapAlloc(heap, size);
+    if (largerCopy == nullptr)
+    {
+        PrivateHeapFree(ptr);
```

Comment:
I think this is free not permitted?

Reply:
You are right - did not closely read realloc return documentation.

Updated.